lib/browser/api/menu.js

```diff
@@ -45,30 +45,49 @@ Menu.prototype._init = function () {
 
 Menu.prototype.popup = function (window, x, y, positioningItem) {
   let [newX, newY, newPosition, newWindow] = [x, y, positioningItem, window]
+  let opts
 
   // menu.popup(x, y, positioningItem)
-  if (!window) {
-    // shift over values
-    if (typeof window !== 'object' || window.constructor !== BrowserWindow) {
-      [newPosition, newY, newX, newWindow] = [y, x, window, null]
-    }
+  if (window != null && !(window instanceof BrowserWindow)) {
+    [newPosition, newY, newX, newWindow] = [y, x, window, null]
   }
 
+  // menu.popup({})
+  if (window && window.constructor !== BrowserWindow) {
```

Comment:
Numbers will pass this check too:
```js
42 && 42..constructor !== BrowserWindow  // true!
```

`(window != null) && (window.constructor === Object)` should work fine.